lib/active_shipping/carriers/usps.rb

```diff
@@ -654,6 +660,71 @@ def parse_tracking_info(response, node)
       )
     end
 
+    def build_address_standardization_request(addresses)
+      xml_builder = Nokogiri::XML::Builder.new do |xml|
+        xml.AddressValidateRequest('USERID' => @options[:login]) do
+        xml.IncludeOptionalElements(true)
+          xml.ReturnCarrierRoute(true)
+          Array(addresses).each_with_index do |address, id|
+            xml.Address('ID' => id) do
+              xml.FirmName(address[:firm_name])
+              xml.Address1(address[:address1])
```

Comment:
From the documentation, it looks like `Address1` and `Address2` are flipped for USPS.

![image](https://cloud.githubusercontent.com/assets/4521864/20768413/de3303cc-b70b-11e6-8d47-2f6e61dddf41.png)

We use [`Address1` as the street address in `ActiveShipping`](https://github.com/Shopify/active_shipping/blob/master/lib/active_shipping/location.rb#L52) but USPS uses it as the suite number, with `Address2` being their street address.